k4FWCore/include/k4FWCore/FunctionalUtils.h

```diff
@@ -127,24 +127,22 @@ namespace k4FWCore {
     template <size_t Index, typename... In, typename... Handles, typename InputTuple>
     void readVectorInputs(const std::tuple<Handles...>& handles, auto thisClass, InputTuple& inputTuple) {
       if constexpr (Index < sizeof...(Handles)) {
-        if constexpr (isVectorLike_v<std::tuple_element_t<Index, std::tuple<In...>>>) {
-          // Bare EDM4hep type, without pointers
-          using EDM4hepType =
-              std::remove_pointer_t<typename std::tuple_element_t<Index, std::tuple<In...>>::value_type>;
-          auto inputMap = std::vector<const EDM4hepType*>();
+        using TupleType = std::tuple_element_t<Index, std::tuple<In...>>;
+        if constexpr (isVectorLike_v<TupleType>) {
+          // Bare EDM4hep type, without pointers or const
+          using EDM4hepType = std::remove_cvref_t<std::remove_pointer_t<typename TupleType::value_type>>;
+          auto inputMap     = std::vector<const EDM4hepType*>();
           for (auto& handle : std::get<Index>(handles)) {
-            if constexpr (std::is_same_v<EDM4hepType, const podio::CollectionBase*>) {
-              inputMap.push_back(&get(handle, thisClass, Gaudi::Hive::currentContext()));
-            } else {
-              podio::CollectionBase* in = handle.get()->get();
-              inputMap.push_back(static_cast<EDM4hepType*>(in));
-            }
+            podio::CollectionBase* in = handle.get()->get();
+            inputMap.push_back(static_cast<const EDM4hepType*>(in));
```

Comment:
I think this answers the question about the CollectionMerger @tmadlener, it seems that both branches are equivalent. I'm not sure how, I guess when I added it it wasn't compiling for the Merger but it's fine to be removed.